### PR TITLE
Removing netstandard configurations for things that are bellow netstandard

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @ericstj 
FYI: @mellinoe @tarekgh 

Removing netstandard configurations to the assemblies that are part of netstandard.dll. The full list I used to base myself is https://github.com/dotnet/standard/blob/master/netstandard/pkg/shims/netstandard.shimlist.txt

I will be adding uap configurations to these same projects on a separate PR

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/src/Configurations.props [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <BuildConfigurations>
       net463-Windows_NT;
       netcoreapp;
-      netstandard;
+      netcoreapp;
```
Comment (mellinoe) [inline src/System.Runtime.Serialization.Formatters/src/Configurations.props]: Duplicate

File: src/System.Runtime.Serialization.Primitives/src/Configurations.props [hunk 2]
```diff
@@ -4,9 +4,9 @@
     <BuildConfigurations>
       netcore50aot-Windows_NT;
       uap101aot-Windows_NT;
-      netstandard1.3;
+      netcoreapp;
       net463-Windows_NT;
-      netstandard;
+      netcoreapp;
```
Comment (mellinoe) [inline src/System.Runtime.Serialization.Primitives/src/Configurations.props]: Duplicate

**Comment by mellinoe** [comment]
LGTM; just a couple of duplicates that I noticed. I trusted that the csproj changes matched up with the config renames.

**Comment by joperezr** [comment]
Looks like 4 S.IO.Compression tests started failing because of this change, although I'm not sure why would that happen. I'll do some debugging to see what is going on.

File: src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj [hunk 3]
```diff
@@ -10,16 +10,16 @@
     <!-- copy to uap10.1 until nuget makes the change to map that to ns1.7 -->
   </PropertyGroup>
   <!-- Help VS understand available configurations -->
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='netstandard-Unix-Debug|AnyCPU'" />
```
Comment (weshaggard) [inline src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj]: Can we also do the find/replace in the matching sln files for these?

File: src/System.IO.Compression/src/System.IO.Compression.csproj [hunk 4]
```diff
@@ -7,15 +7,15 @@
     <ProjectGuid>{5471BFE8-8071-466F-838E-5ADAA779E742}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <UseECMAKey>true</UseECMAKey>
-    <DefineConstants Condition="'$(TargetGroup)' == 'netstandard'">$(DefineConstants);FEATURE_ZLIB</DefineConstants>
+    <DefineConstants Condition="'$(TargetGroup)' == 'netcoreapp'">$(DefineConstants);FEATURE_ZLIB</DefineConstants>
     <DefineConstants>$(DefineConstants);netstandard20</DefineConstants>
```
Comment (weshaggard) [inline src/System.IO.Compression/src/System.IO.Compression.csproj]: Do we need this #ifdef any longer?

File: src/System.IO.Compression/src/System.IO.Compression.csproj [hunk 5]
```diff
@@ -8,7 +8,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <UseECMAKey>true</UseECMAKey>
     <DefineConstants Condition="'$(TargetGroup)' == 'netcoreapp'">$(DefineConstants);FEATURE_ZLIB</DefineConstants>
-    <DefineConstants>$(DefineConstants);netcoreapp20</DefineConstants>
+    <DefineConstants>$(DefineConstants);netstandard20</DefineConstants>
```
Comment (weshaggard) [inline src/System.IO.Compression/src/System.IO.Compression.csproj]: I would remove this #ifdef given it isn't actually conditioned at all in this project.

File: src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj (hunk 3 above)
Comment (joperezr) [inline src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj]: yes, I'll fix this

File: src/System.IO.Compression/src/System.IO.Compression.csproj (hunk 4 above)
Comment (joperezr) [inline src/System.IO.Compression/src/System.IO.Compression.csproj]: netstandard20 one or the FEATURE_ZLIB one?

File: src/System.IO.Compression/src/System.IO.Compression.csproj [hunk 6]
```diff
@@ -7,8 +7,6 @@
     <ProjectGuid>{5471BFE8-8071-466F-838E-5ADAA779E742}</ProjectGuid>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <UseECMAKey>true</UseECMAKey>
-    <DefineConstants Condition="'$(TargetGroup)' == 'netcoreapp'">$(DefineConstants);FEATURE_ZLIB</DefineConstants>
```
Comment (weshaggard) [inline src/System.IO.Compression/src/System.IO.Compression.csproj]: Is FEATURE_ZLIB no longer used anywhere?

Comment (joperezr) [inline src/System.IO.Compression/src/System.IO.Compression.csproj]: It was, Crc32Helper.cs but that is why I deleted the code on that was no longer applicable on that file. Changes to that file are on commit 44d9f7f

**Comment by joperezr** [comment]
Merging now. Will address any extra feedback in next PR.

**Pull request merged**